### PR TITLE
(#1687512) (#1687512) core: imply NNP and SUID/SGID restriction for DynamicUser=yes service

### PR DESCRIPTION
**Pull request by jsynacek**
Let's be safe, rather than sorry. This way DynamicUser=yes services can
neither take benefit of, nor create SUID/SGID binaries.

Given that DynamicUser= is a recent addition only we should be able to
get away with turning this on, even though this is strictly speaking a
binary compatibility breakage.

(cherry picked from commit bf65b7e0c9fc215897b676ab9a7c9d1c688143ba)
Resolves: #1687512

### CODE REVIEW

**Comment by jsynacek** [comment]
v2.

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -1747,3 +1749,174 @@ int seccomp_lock_personality(unsigned long personality) {
 
         return 0;
 }
+
+int seccomp_protect_hostname(void) {
```
Comment (msekletar) [inline src/shared/seccomp-util.c]: This function was probably backported by accident. 

I am fine with including it but we also need to backport corresponding parser, man page blurp, etc ...

**Review by msekletar — CHANGES_REQUESTED** [review]
LGTM, except the ProtectHostname= problem.

File: src/shared/seccomp-util.c (hunk 1 above)
Comment (jsynacek) [reply src/shared/seccomp-util.c]: Right, this was by accident. I'm going to remove it.

**Comment by jsynacek** [comment]
v3.

**Review by msekletar — APPROVED** [review]
LGTM

**Pull request merged**